### PR TITLE
Integrate with new gossipsub version

### LINKED ISSUE #643: override default libp2p pubsub message-id to be content-addressed
v0.9.4: (ethereum/eth2.0-specs#1538)

### PR DESCRIPTION
**Pull request by twoeths**
## Goal
+ Close #643

### CODE REVIEW

**Review by mpetrunic — CHANGES_REQUESTED** [review]
Could you apply changes to branch with regular sync. Most of stuff is already there, I think only message cache might nee to be updated

**Review by wemeetagain — DISMISSED** [review]

File: packages/lodestar/src/network/gossip/gossipsub.ts [hunk 1]
```diff
@@ -1,20 +1,21 @@
...
   getGossipTopic,
-  getMessageId,
   getSubnetFromAttestationSubnetTopic,
   isAttestationSubnetTopic,
   normalizeInRpcMessage
 } from "./utils";
 import {GossipEvent} from "./constants";
 import {GOSSIP_MAX_SIZE} from "../../constants";
 
+const msgIdFn = (message: IGossipMessage): string => Buffer.from(hash(message.data)).toString("base64");
```
Comment (mpetrunic) [inline packages/lodestar/src/network/gossip/gossipsub.ts:17]: We already have that, it's called getMessageId

File: packages/lodestar/src/network/gossip/gossipsub.ts [hunk 2]
```diff
@@ -58,7 +59,7 @@ export class LodestarGossipsub extends Gossipsub {
     assert(message.data.length <= GOSSIP_MAX_SIZE, `Message exceeds size limit of ${GOSSIP_MAX_SIZE} bytes`);
     const topic = message.topicIDs[0];
     // avoid duplicate
-    if (this.transformedObjects.get(message.messageId)) {
+    if (this.transformedObjects.get(msgIdFn(message))) {
```
Comment (mpetrunic) [inline packages/lodestar/src/network/gossip/gossipsub.ts:62]: I've extended IGossipMessage in normalizeInRpcMessage to include messageId

**Review by mpetrunic — CHANGES_REQUESTED** [review]
I think you just need to pass getMessageId function to super constructor options

**Comment by twoeths** [comment]
Replaced by #754

**Pull request closed without merging**